pycqed/instrument_drivers/physical_instruments/_QCC/QCCMicrocode.py

```diff
@@ -0,0 +1,284 @@
+
+        if present_format is True:
+            bin_str = get_bin(condition, 2) + "|"
+            bin_str += get_bin(op_type_left, 2) + " "
+            bin_str += get_bin(cw_left, 8) + "|"
+            bin_str += get_bin(op_type_right, 2) + " "
+            bin_str += get_bin(cw_right, 8)
+            bin_str = "0"*10 + bin_str
+
+        return final_val
+
+    def disa_cs_line(self, int_data):
+        if not isinstance(int_data, int):
+            raise ValueError(
+                "The input data {} is not an integer".format(int_data))
+
+        condition = (int_data >> 20) & 3
+        op_type_left = (int_data >> 18) & 3
+        cw_left = (int_data >> 10) & 0xFF
+        op_type_right = (int_data >> 8) & 3
+        cw_right = int_data & 0xFF
+        return condition, op_type_left, cw_left, op_type_right, cw_right
+
+    def disa_bin_microcode(self, microcode):
+        cs_line_array = []
+        if (len(microcode) % 4) != 0:
+            raise ValueError("The input binary microcode should be a"
+                             " multiple of 4.")
+
+        for i in range(int(len(microcode) / 4)):
+            cs_line_array.append(int.from_bytes(
+                microcode[i*4: i*4 + 4], byteorder='little', signed=False))
+
+        return cs_line_array
+
+    def print_pure_cs_line(self, cs_line):
+        (condition, op_type_left, cw_left, op_type_right, cw_right) = \
+            self.disa_cs_line(cs_line)
+        print(self.CS_format.format(condition, op_type_left, cw_left,
+                                    op_type_right, cw_right))
+
+    def print_compact_cs_line(self, cs_line):
+        (condition, op_type_left, cw_left, op_type_right, cw_right) = \
+            self.disa_cs_line(cs_line)
+
+        print("{:3d} {:6s} ".format(cw_left, "(" +
+                self.type_dict[op_type_left] + ")"),
+                end = '')
+        if op_type_right != 0:
+            print("{:3d} {:6s}  ".format(cw_right, "(" +
+                self.type_dict[op_type_right] + ")"),
+                end = '')
+        if condition != 0:
+            print("condition: {}".format(condition), end='')
+
+    def print_cs_line(self, line_number):
+        print(self.CS_header)
+        self.print_pure_cs_line(self.microcode[line_number])
+
+    def print_cs_line_no_header(self, line_number):
+        self.print_compact_cs_line(self.microcode[line_number])
+
+    def dump_microcode(self, filename=None):
+        if filename is not None:
+            try:
+                mc_config = open(filename, 'w', encoding='utf-8')
+                logging.info("opened file {} successfully.".format(filename))
+            except:
+                raise OSError('\tError: Failed to open file ' +
+                              self.filename + ".")
+            # direct std output into the write file
+            saveout = sys.stdout
+            sys.stdout = mc_config
+
+        print("     ", self.CS_header)
```

Comment:
Not sure if this has any value to the user

Reply:
Maintaining to provide similar behavior upon initialization, as expected when starting CC-Light